### ISSUE TITLE
Progress Bar support

### ISSUE THREAD

**Issue by lukelex** [Toqozz/wired-notify#4]
Any plans on adding support for a progress bar?

**Comment by Toqozz** [comment]
A progress bar block should be relatively trivial to support.

Would you be able to elaborate more on your use case?  I know dunst has this, but I'm unclear on the specifics, or if you want anything different to what they have over there.

From what I can tell, with dunst, you normally send a notification with a progress hint; `notify-send -h int:value:42 "Working ..."`, and then the notification displays some progress bar if that hint is present.

We'd probably want to use [`render_criteria`](https://github.com/Toqozz/wired-notify/wiki/Blocks#render-criteria) to decide when to draw the progress bar, and just have a block that reads that value and shows it.  You'd have to replace the notification using its ID if you wanted to update a volume slider or something like that, which I don't think `notify-send` supports.  `notify-call` does support that but it's less prominent.

Would simply a block like that obeys a hint satisfy you, or is there anything else you want to add?

The only thing I'm wanting to avoid is making a custom `wired-send` tool or something that supports custom functionality.  I think `wired` should just be a notify receive tool and other things are better suited to sending notifications, like `notify-call`.

Apologies for the wall of text.

**Comment by lukelex** [comment]
![image](https://user-images.githubusercontent.com/842850/128477668-8f2a9385-8c45-4d95-88c3-af5c08b44abe.png)

Dunst doesn't support styling the progress bar and renders it in a fixed position. This kinda defeats the purpose, for my use case, as I'm trying to have it blend with the body of the notification.

I should be able to place the notification wherever I'd like to, whether that's the body or header.

For example, I'd live the bar volume to be in body:
![image](https://user-images.githubusercontent.com/842850/128479136-5f17c0a1-ad42-403f-b67b-91952ce4e2c4.png)

**Comment by Toqozz** [comment]
This is available on master.  You can add a progress block to your config with the name `ProgressBlock`:
```rust
        (
            name: "progress",
            parent: "body",
            hook: Hook(parent_anchor: BL, self_anchor: TL),
            offset: Vec2(x: 0.0, y: 0.0),
            params: ProgressBlock((
                padding: Padding(left: 3.0, right: 3.0, top: 3.0, bottom: 3.0),
                border_width: 3.0,
                border_rounding: 10.0,
                border_color: Color(hex: "#ebdbb2"),
                background_color: Color(hex: "#ff0000"),
                fill_color: Color(hex: "#00ff00"),
                size: Vec2(x: 100.0, y: 30.0), // x: -1.0, and y: -1.0 will match parent rect size.
                border_color_hovered: Color(hex: "#fbf1c7"),
                background_color_hovered: Color(hex: "#3c3836"),
                fill_color_hovered: Color(hex: "#fbf1c7"),
            )),
        ),
```

It reads the the `value` hint, which you can send like the following:

`notify-send -h int:value:<progress> "Working..." "I'm working on something."`

Where `<progress>` is an integer from 0-100.

If you get a chance to try it, let me know if you come across any issues or things you would like changed.

**Comment by Toqozz** [comment]
Released in [0.9.2](https://github.com/Toqozz/wired-notify/releases/tag/0.9.2).
